### PR TITLE
Add various effective UID and groups operations

### PR DESCRIPTION
**Pull request by jez9999**
Addresses issue #21114.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -477,6 +477,14 @@ declare namespace NodeJS {
         getuid(): number;
         setuid(id: number): void;
         setuid(id: string): void;
+        geteuid(): number;
+        seteuid(id: number): void;
+        seteuid(id: string): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: Please use an union type instead an overload, e.a. `seteuid(id: string | number): void`.
Would be nice if you could adapt the nearby functions `setgid` and `setuid` also...

File: types/node/index.d.ts [hunk 2]
```diff
@@ -477,6 +477,14 @@ declare namespace NodeJS {
...
+        getegid(): number;
+        setegid(id: number): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: Same as for `seteuid`, please use an union type instead an overload.

File: types/node/index.d.ts [hunk 3]
```diff
@@ -477,6 +477,14 @@ declare namespace NodeJS {
...
+        setegid(id: string): void;
+        getgroups(): number[];
+        setgroups(groups: (string | number)[]): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think you have to write `Array<string | number>` to keep lint happy.

**Review by Flarna — CHANGES_REQUESTED** [review]

File: types/node/index.d.ts (hunk 1 above)
Comment (jez9999) [reply types/node/index.d.ts]: Why is a union type better than an overload?

**Comment by RyanCavanaugh** [comment]
@jez9999 Please fix the failures indicated in the Travis CI log.

File: types/node/index.d.ts (hunk 1 above)
Comment (Flarna) [reply types/node/index.d.ts]: It's the way how typescript does the lookup and type detection. If you use an union type it's one function with one signature, if you use overloads then they are two functions.
Following does not transpile currently but works if union types are used:
```
function foo<T>(f: (p: T) => void, p: T) {
    return f(p);
}
foo(process.setgid, "foo"); // OK
foo(process.setgid, 5); // Argument of type '5' is not assignable to parameter of type 'string'
```
Typescript select always the last overload in such situations.

File: types/node/index.d.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@
 //                 Deividas Bakanas <https://github.com/DeividasBakanas>
 //                 Kelvin Jin <https://github.com/kjin>
 //                 Alvis HT Tang <https://github.com/alvis>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped)
```
Comment (Flarna) [inline types/node/index.d.ts]: I expect this change was not intended.

**Comment by jez9999** [comment]
I changed it to unions.

**Review by Flarna — APPROVED** [review]

**Comment by jez9999** [comment]
@Flarna @RyanCavanaugh Can these be merged now?

**Comment by Flarna** [comment]
@jez9999 I have no write access in this reqo so I can't help here. My part was done by reviewing the PR. I know that contributing to node typings is cumbersome currently as CI fails usually and as a result PRs are not merged for quite some time. see #20308

**Comment by jez9999** [comment]
@DanielRosenwasser Can this be merged please?

**Comment by jez9999** [comment]
I don't think that failure has anything to do with my changes.

**Comment by jez9999** [comment]
@andy-ms @ianfp @DanielRosenwasser @andypyrope @evanshortiss Somebody please merge this in!  Otherwise my code using Process breaks unless i hack it temporarily.  And this project needs more people reviewing/checking stuff in :-\

**Comment by evanshortiss** [comment]
@jez9999 not sure why you've mentioned me, I'm not an org member with merge rights. Maybe a mistake?

**Comment by jez9999** [comment]
It said you merged this: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/02fc61f4ad54e74bbc70fc8d00d0e139cb5a6ba7

**Comment by evanshortiss** [comment]
@jez9999 I see, I made the PR yep - most folks [here](https://github.com/orgs/DefinitelyTyped/people) so you could reference them by typing `@DefinitelyTyped/DefinitelyTyped` - a trick I learned recently 😄

**Comment by jez9999** [comment]
Luckily it's been merged now.

**Pull request merged**